src/components/Modals/Subjects/Subjects.js

```diff
@@ -99,7 +96,7 @@ export default function Subjects ({
             const char = i === subjectIndex ? `\u25CF` : `\u25CB`
             return (
               <Button
-                key={`SUBJECTS_${subj.id}`}
+                key={`SUBJECTS_PAGE_${i}`}
```

Comment:
Using the subject id here would be would be preferred as array indexes are unreliable for the use as a key in react renders since they aren't guaranteed to be referencing the same item each time or be unique.

Reply:
@srallen I totally agree, but when I tried to use `subj.id` as a key, I found that some of the fetched subject's `id` property is `undefined`. For example, when I draw a rectangle fetching 600 subjects about 10% don't have an `id`, so I'll make a note to find out why, but for now use array index.

Reply:
Oh yeah, that is a problem. Possibly with the original data?

Reply:
This is surprising to me; that ID should exist for all subjects, so I agree with Sarah that this is probably an issue with the data.  I'll take a quick look at data in the DB and report back.

Reply:
I'm not seeing anything unusual `id`s in the DB `subjects` table, but maybe it's getting mangled by the AzFunc?

Reply:
@goplayoutside3 If you can flag an example of a case where subj.id is undefined, that might help us diagnose what might be happening.

Reply:
I found that this key identifies a chunk of subjects, so each button component actually maps from an array of 9 subjects. It does not correspond to a subject object fetched from the database. If the index of the chunked array of subjects doesn't work here, I could use the `id` of the first subject of the array chunk?

Reply:
Ah, that makes sense. I didn't catch that it was the chunked subjects that was being iterated through either initially. We're chunking because we're creating pagination here, so maybe the nested array after chunking could be converted into a [map](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map) and the key would be the page number? This is pretty low priority I think, so maybe something to do after other higher priority work.

Reply:
I just realized you'd have to convert the map back to an array to call `map`  on it, so maybe not.  :)